src/include/ci/net/ipvx.h

```diff
@@ -324,6 +324,12 @@ typedef union ci_ipx_hdr {
 #endif
 } ci_ipx_hdr_t;
 
+#if CI_CFG_IPV6
+#define ipx_hdr_ptr(af, ipx) (af == AF_INET6 ? &ipx->ip6 : &ipx->ip4)
```

Comment:
I ran a quick test and discovered a built-time issue:
```console
$ TRANSPORT_CONFIG_OPT_HDR=ci/internal/transport_config_opt_cloud.h make -j16
...
src/include/ci/net/ipvx.h:328:58: error: pointer type mismatch in conditional expression [-Werror]
 #define ipx_hdr_ptr(af, ipx) (af == AF_INET6 ? &ipx->ip6 : &ipx->ip4)
                                                          ^
build/x86_64_linux-4.18.0-372.9.1.el8.x86_64/lib/transport/ip/netif_event.c:101:19: note: in expansion of macro ‘ipx_hdr_ptr’
   void *ipx_hdr = ipx_hdr_ptr(af, ipx);
                   ^~~~~~~~~~~
```

Could you apply the type-casting and see if it works for you?
```suggestion
#define ipx_hdr_ptr(af, ipx) \
    (af == AF_INET6 ? (void*)&ipx->ip6 : (void*)&ipx->ip4)
```

Reply:
Thanks! Fixed.